components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/CoreListener.java

```diff
@@ -0,0 +1,270 @@
+package org.wso2.carbon.inbound.endpoint.protocol.nats;
+
+import io.nats.client.Connection;
+import io.nats.client.Options;
+import io.nats.client.Nats;
+import io.nats.client.Subscription;
+import io.nats.client.Message;
+import net.minidev.json.JSONUtil;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.KeyManagementException;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.Properties;
+
+public class CoreListener implements NatsMessageListener {
+    private static final Log log = LogFactory.getLog(CoreListener.class.getName());
+    private String subject;
+    private NatsInjectHandler injectHandler;
+    private Properties natsProperties;
+    private Connection connection;
+
+    public CoreListener(String subject, NatsInjectHandler injectHandler, Properties natsProperties) {
+        this.subject = subject;
+        this.injectHandler = injectHandler;
+        this.natsProperties = natsProperties;
+    }
+
+    @Override public boolean createConnection() throws IOException, InterruptedException {
+        if (connection == null) {
+            connection = getNatsConnection();
```

Comment:
The method names getNatsConnection and createConnection seem to be contradicting